script.js

```diff
@@ -0,0 +1,959 @@
+        {
+          continue;
+        }
+        indexThreeOfAKindFound[i] = true;
+        indexThreeOfAKindFound[j] = true;
+        indexThreeOfAKindFound[k] = true;
+        return { match: true, indices: indexThreeOfAKindFound };
+      } // third loop end
+    } // second loop end
+  }
+  return { match: false, indices: indexThreeOfAKindFound };
+};
+
+// 'Straight', 'Five consecutive cards of mixed suits;
+// for example, 2 of diamonds, 3 of hearts, 4 of diamonds, 5 of clubs, 6 of spades.'
+const isStraight = () => {
+  let previousCardRank = 0;
+  let isStraightCards = true;
+  for (let i = 0; i < boardOfDealtCards.length; i += 1)
+  {
+    const currentCardRank = boardOfDealtCards[i].rank;
+    if (i === 0)
+    {
+      // If the first card is Ace, take it's rank as 14 and compare
+      previousCardRank = (boardOfDealtCards[i].display === ACE_DISPLAY_NAME)
+        ? 14 : currentCardRank;
+      continue;
+    }
+    if ((previousCardRank - currentCardRank) !== 1)
+    {
+      isStraightCards = false;
+      break;
+    }
+  }
+  return isStraightCards;
+};
+
+// 'Flush', 'Five cards of the same suit;
+// for example, ace, 10, 7, 4, 3, all of diamonds.'
+const isFlush = () => {
+  let prevoiusCardSuit = '';
+  // boardOfDealtCards = TestArray2;
+  for (let i = 0; i < boardOfDealtCards.length; i += 1)
+  {
+    const currentCardSuit = boardOfDealtCards[i].suit;
+    if (i === 0)
+    {
+      prevoiusCardSuit = currentCardSuit;
+      continue;
+    }
+    // If any card suit is different from others, return false
+    if (!(prevoiusCardSuit === currentCardSuit))
+    {
+      return false;
+    }
+    prevoiusCardSuit = currentCardSuit;
+  }
+  return true;
+};
+
+// 'Full house', 'Three cards of one rank, two cards of another rank;
+// for example, 3 of diamonds, 3 of hearts, 3 of spades, 6 of hearts, 6 of spades.'
+const isFullHouse = () => {
+  // First check whether there are 3 cards of same kind.
+  const resThreeKind = isThreeOfAKind();
+  if (!resThreeKind.match) {
+    // If three of a kind match is not found return false.
+    return false;
+  }
+  // Filter out all the values that are not having value as "true".
+  // This will give the index of the remaining 2 (?) cards that are to be verified
+  const remIndexArray = resThreeKind.indices.filter((index) => (index !== true));
+
+  // If 3 same kind cards are found, check for the other pair
+  let tempCard = null;
+  for (let i = 0; i < remIndexArray.length; i += 1)
+  {
+    if (tempCard === null) {
+      tempCard = boardOfDealtCards[remIndexArray[i]];
+    }
+    else { // compare both the card elements
+      const nextCard = boardOfDealtCards[remIndexArray[i]];
+      if (nextCard.rank === tempCard.rank) {
+        // It's a pair.
+        return true;
+      }
+    }
+  }
+  return false;
+};
+
+// 'Four of a kind', 'Four cards of the same rank;
+// for example, ace of hearts, ace of spades, ace of clubs, ace of diamonds.'
+const isFourOfAKind = () => {
```

Comment:
Can we achieve this with a single for loop instead of 4 nested for loops?

Reply:
Tried..But not enough time to test all. So keeping the same code for time being. Will look into it later